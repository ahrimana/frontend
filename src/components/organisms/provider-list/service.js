import axios from 'axios'
import { API_ROOT } from 'utils'


async function providerList(page) {
  const response = await axios.get(
    `${API_ROOT}/provider`,
    { headers: { 'X-Page': page } },
  )
  return response.data
}


export default { providerList }
