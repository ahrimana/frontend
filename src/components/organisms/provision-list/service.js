import axios from 'axios'
import { API_ROOT } from 'utils'


async function provisionList(page) {
  const response = await axios.get(
    `${API_ROOT}/provision`,
    { headers: { 'X-Page': page } },
  )
  return response.data
}


export default { provisionList }
