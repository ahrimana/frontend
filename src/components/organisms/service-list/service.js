import axios from 'axios'
import { API_ROOT } from 'utils'


async function serviceList(page) {
  const response = await axios.get(
    `${API_ROOT}/service`,
    { headers: { 'X-Page': page } },
  )
  return response.data
}


export default { serviceList }
