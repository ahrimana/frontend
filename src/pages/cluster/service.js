import axios from 'axios'
import { API_ROOT } from 'utils'


async function fetchAll(page) {
  const response = await axios.get(
    `${API_ROOT}/cluster`,
    { headers: { 'X-Page': page } },
  )
  return response.data
}


export default {
  fetchAll,
}
