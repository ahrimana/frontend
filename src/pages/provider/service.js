import axios from 'axios'
import { API_ROOT, getCookie } from 'utils'


async function create(name) {
  const csrf = getCookie('csrf_access_token')
  const response = await axios.post(
    `${API_ROOT}/provider`,
    { name },
    {
      headers: {
        'X-CSRF-TOKEN': csrf,
      },
    },
  )
  return response.data
}


async function edit(provider, data) {
  const csrf = getCookie('csrf_access_token')
  const response = await axios.patch(
    `${API_ROOT}/provider/${provider.id}`,
    data,
    {
      headers: {
        'X-CSRF-TOKEN': csrf,
      },
    },
  )
  return response.data
}


async function fetchAll(page) {
  const csrf = getCookie('csrf_access_token')
  const response = await axios.get(
    `${API_ROOT}/provider`,
    {
      headers: {
        'X-Page': page,
        'X-CSRF-TOKEN': csrf,
      },
    },
  )
  return response.data
}


async function fetch(id) {
  const csrf = getCookie('csrf_access_token')
  const response = await axios.get(
    `${API_ROOT}/provider/${id}`,
    {
      headers: { 'X-CSRF-TOKEN': csrf },
    },
  )
  return response.data
}


async function remove(id) {
  const csrf = getCookie('csrf_access_token')
  const response = await axios.delete(
    `${API_ROOT}/provider/${id}`,
    {
      headers: {
        'X-CSRF-TOKEN': csrf,
      },
    },
  )
  return response.data
}


export default {
  create,
  edit,
  fetch,
  fetchAll,
  remove,
}
