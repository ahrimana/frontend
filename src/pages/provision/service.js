import axios from 'axios'
import { API_ROOT } from 'utils'


async function fetchAll(page) {
  const response = await axios.get(
    `${API_ROOT}/provision`,
    { headers: { 'X-Page': page } },
  )
  return response.data
}


export default {
  fetchAll,
}
