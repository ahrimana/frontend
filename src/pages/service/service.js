import axios from 'axios'
import { API_ROOT, getCookie } from 'utils'


async function create(name) {
  const csrf = getCookie('csrf_access_token')
  const response = await axios.post(
    `${API_ROOT}/service`,
    { name },
    {
      headers: {
        'X-CSRF-TOKEN': csrf,
      },
    },
  )
  return response.data
}


async function edit(service, data) {
  const csrf = getCookie('csrf_access_token')
  const response = await axios.patch(
    `${API_ROOT}/service/${service.id}`,
    data,
    {
      headers: {
        'X-CSRF-TOKEN': csrf,
      },
    },
  )
  return response.data
}


async function fetchAll(page) {
  const csrf = getCookie('csrf_access_token')
  const response = await axios.get(
    `${API_ROOT}/service`,
    {
      headers: {
        'X-Page': page,
        'X-CSRF-TOKEN': csrf,
      },
    },
  )
  return response.data
}


async function fetch(id) {
  const csrf = getCookie('csrf_access_token')
  const response = await axios.get(
    `${API_ROOT}/service/${id}`,
    {
      headers: { 'X-CSRF-TOKEN': csrf },
    },
  )
  return response.data
}


async function remove(id) {
  const csrf = getCookie('csrf_access_token')
  const response = await axios.delete(
    `${API_ROOT}/service/${id}`,
    {
      headers: {
        'X-CSRF-TOKEN': csrf,
      },
    },
  )
  return response.data
}


export default {
  create,
  edit,
  fetch,
  fetchAll,
  remove,
}
