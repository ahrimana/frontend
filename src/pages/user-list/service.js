import axios from 'axios'
import { API_ROOT } from 'utils'


async function userList() {
  const response = await axios.get(`${API_ROOT}/users`)
  return response.data
}


export default { userList }
