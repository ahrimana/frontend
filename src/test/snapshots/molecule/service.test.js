/* eslint no-undef: 0 */
/* eslint no-unused-expressions: 0 */
/* eslint no-sequences: 0 */
/* eslint quotes: 0 */

import React from 'react';
import { Service } from '../../../components/molecules/service';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe("Service Molecule (snapshot)", () => {
  it("Service molecule render paper", () => {
    const component = shallow(
      <Service
        name="vagrant"
        serviceId="1337testId"
      />
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
