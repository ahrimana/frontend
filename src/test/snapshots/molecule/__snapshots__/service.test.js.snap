exports[`Service Molecule (snapshot) Service molecule render paper 1`] = `
<Paper
  circle={false}
  rounded={true}
  style={
    Object {
      "borderRadius": 10,
      "display": "inline-block",
      "height": 120,
      "margin": 3,
      "textAlign": "center",
      "width": 120,
    }
  }
  transitionEnabled={true}
  zDepth={1}>
  <div
    onClick={[Function]}
    style={
      Object {
        "marginLeft": 100,
      }
    }>
    x
  </div>
  <ActionBuild
    color="#ccc"
    style={
      Object {
        "color": "#ccc",
        "height": "60%",
        "width": "100%",
      }
    } />
  <div>
    vagrant
  </div>
</Paper>
`;
